end_to_end_tests/generated_code_live_tests/test_enum_and_const.py

```diff
@@ -0,0 +1,204 @@
+        with pytest.raises(ValueError):
+            MyModel.from_dict({"intEnumProp": "a"})
+
+
+@with_generated_client_fixture(
+"""
+components:
+  schemas:
+    MyEnum:
+      type: string
+      enum: ["a", "A"]
+    MyIntEnum:
+      type: integer
+      enum: [2, 3]
+    MyModel:
+      properties:
+        enumProp: {"$ref": "#/components/schemas/MyEnum"}
+        intEnumProp: {"$ref": "#/components/schemas/MyIntEnum"}
+        nullableEnumProp:
+          oneOf:
+            - {"$ref": "#/components/schemas/MyEnum"}
+            - type: "null"
+""",
+    config="""
+literal_enums: true
+""",
+)
+@with_generated_code_import(".models.MyModel")
+class TestLiteralEnums:
+    def test_enum_prop(self, MyModel):
+        assert_model_decode_encode(MyModel, {"enumProp": "a"}, MyModel(enum_prop="a"))
+        assert_model_decode_encode(MyModel, {"enumProp": "A"}, MyModel(enum_prop="A"))
+        assert_model_decode_encode(MyModel, {"intEnumProp": 2}, MyModel(int_enum_prop=2))
+
+    def test_enum_prop_type(self, MyModel):
+        assert MyModel.from_dict({"enumProp": "a"}).enum_prop.__class__ is str
+        assert MyModel.from_dict({"intEnumProp": 2}).int_enum_prop.__class__ is int
+
+    def test_nullable_enum_prop(self, MyModel):
+        assert_model_decode_encode(
+            MyModel,
+            {"nullableEnumProp": "a"},
+            MyModel(nullable_enum_prop="a"),
+        )
+        assert_model_decode_encode(
+            MyModel,
+            {"nullableEnumProp": None},
+            MyModel(nullable_enum_prop=None),
+        )
+
+    def test_invalid_values(self, MyModel):
+        with pytest.raises(TypeError):
+            MyModel.from_dict({"enumProp": "c"})
+        with pytest.raises(TypeError):
+            MyModel.from_dict({"enumProp": 2})
+        with pytest.raises(TypeError):
+            MyModel.from_dict({"intEnumProp": 0})
+        with pytest.raises(TypeError):
+            MyModel.from_dict({"intEnumProp": "a"})
+
+
+@with_generated_client_fixture(
+"""
+components:
+  schemas:
+    MyModel:
+      properties:
+        mustBeErnest:
+          const: Ernest
+        mustBeThirty:
+          const: 30
+""",
+)
+@with_generated_code_import(".models.MyModel")
+class TestConst:
+    def test_valid_string(self, MyModel):
+        assert_model_decode_encode(
+            MyModel,
+            {"mustBeErnest": "Ernest"},
+            MyModel(must_be_ernest="Ernest"),
+        )
+
+    def test_valid_int(self, MyModel):
+        assert_model_decode_encode(
+            MyModel,
+            {"mustBeThirty": 30},
+            MyModel(must_be_thirty=30),
+        )
+
+    def test_invalid_string(self, MyModel):
+        with pytest.raises(ValueError):
+            MyModel.from_dict({"mustBeErnest": "Jack"})
+
+    def test_invalid_int(self, MyModel):
+        with pytest.raises(ValueError):
+            MyModel.from_dict({"mustBeThirty": 29})
+
+
+class TestEnumAndConstInvalidSchemas:
```

Comment:
These tests make some of the current low-level unit tests in `test_enum_property.py`, and all of `test_const.py`, unnecessary in terms of coverage.